src/ui/SimpleFilter/SimpleFilter.ts

```diff
@@ -0,0 +1,218 @@
+  values: string[];
+  field: IFieldOption;
+  valueCaption: any;
+}
+
+export class SimpleFilter extends Component {
+  static ID = 'SimpleFilter';
+  static doExport = () => {
+    exportGlobally({
+      'SimpleFilter': SimpleFilter
+    });
+  };
+  static options: ISimpleFilterOptions = {
+
+    values: ComponentOptions.buildListOption<string>(),
+    field: ComponentOptions.buildFieldOption({ required: true }),
+    title: ComponentOptions.buildStringOption({ defaultValue: l('NoTitle') }),
+    valueCaption: ComponentOptions.buildJsonOption()
+  };
+
+  public checkboxes: { checkbox: Checkbox, value: string }[] = [];
+  public checkboxContainer: Dom;
+  private expanded = false;
+  private circleElement: Dom;
+
+  constructor(public element: HTMLElement, public options: ISimpleFilterOptions, public bindings?: IComponentBindings) {
+    super(element, SimpleFilter.ID, bindings);
+    this.options = ComponentOptions.initComponentOptions(element, SimpleFilter, options);
+
+    const select = $$('span', { className: 'coveo-simplefilter-select' });
+    const selectText = $$('span', {className: 'coveo-simplefilter-selecttext'}, l(this.options.title));
+    const icon = $$('span', { className: 'coveo-icon' });
+
+    this.checkboxContainer = $$('div', { className: 'coveo-checkbox-container' });
+    this.checkboxes = _.map(this.options.values, (value) => this.createCheckboxes(value));
+    _.each(this.checkboxes, (result) => {
+      this.checkboxContainer.el.appendChild(result.checkbox.getElement());
+    });
+
+    select.el.appendChild(selectText.el);
+    select.el.appendChild(this.buildCircleElement().el);
+    select.el.appendChild(icon.el);
+    this.element.appendChild(select.el);
+    this.element.appendChild(this.checkboxContainer.el);
+    this.createWrapper().append(this.element);
+
+    $$(this.element).on('click', (e: Event) => this.handleClick(e));
+    this.bind.onRootElement(QueryEvents.buildingQuery, (args: IBuildingQueryEventArgs) => this.handleBuildingQuery(args));
+    this.bind.onRootElement(QueryEvents.querySuccess, (args: IQuerySuccessEventArgs) => this.groupBy(args));
+    this.bind.onRootElement(QueryEvents.doneBuildingQuery, (args: IDoneBuildingQueryEventArgs) => this.handleDoneBuildingQuery(args));
+  }
+
+  private handleBuildingQuery(args: IBuildingQueryEventArgs) {
+    Assert.exists(args);
+    Assert.exists(args.queryBuilder);
+    const selectedValues = this.getSelectedValues();
+    if (selectedValues.length > 0) {
+      args.queryBuilder.advancedExpression.addFieldExpression(this.options.field.toLocaleString(), '==', selectedValues);
+    }
+  }
+
+  public toggle() {
+   this.expanded = $$(this.checkboxContainer).hasClass('coveo-checkbox-container-expanded');
+    if (!this.expanded) {
+      this.open();
+    } else if (this.expanded){
+      this.close();
+    }
+  }
+
+  private handleClick(e: Event) {
+    if (e.target == this.element || e.target == $$(this.element).find('.coveo-simplefilter-select') ||
```

Comment:
I'm wondering if there isn't a better way to do this. Maybe you could check for elements that you don't want to trigger the click instead? At the very least this should be properly aligned.